### PR TITLE
 plan, executor: build logical plan to check column name validation when doPrepare

### PR DESCRIPTION
**Pull request by XuHuaiyu**
After removing resolver.go, TiDB do not check column name validation in preprocess phase.

So, for sql like:
``` sql
create table t(a int);
prepare stmt from 'select b from t';
```
TiDB cannot check the validation of `t.b` until `execute stmt`.

This commit builds logical plan when `doPrepare` using fake values to replace 
the paramMarkers, so the logical plan can not be reused when `execute stmt`, 
we need to rebuild the logical plan again.
PTAL @hanfei1991 @coocood

### CODE REVIEW

**Comment by shenli** [comment]
@XuHuaiyu Please fix CI.

**Comment by hanfei1991** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
PTAL @coocood

File: plan/optimizer.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func Optimize(ctx context.Context, node ast.Node, is infoschema.InfoSchema) (Pla
 	return p, nil
 }
 
-// BuildLogicalPlan is exported and only used for test.
+// BuildLogicalPlan used to build logical plan from ast.Node.
 func BuildLogicalPlan(ctx context.Context, node ast.Node, is infoschema.InfoSchema) (Plan, error) {
```
Comment (zz-jason) [inline plan/optimizer.go]: we can call `BuildLogicalPlan(...)` in `Optimize` now.

**Pull request merged**